### PR TITLE
Delete default token

### PR DESCRIPTION
**Pull request by micafer**

### CODE REVIEW

**Comment by Sftobias** [comment]
I've tried the roles on a new deployment and they don't seem to work correctly. The problem is on the consul server. The rest of the nodes appear to see each other, while the consul server cannot see any of them

method=GET url=/v1/operator/raft/configuration from=127.0.0.1:55712 error="Permission denied: anonymous token lacks permission 'operator:read'. The anonymous token is used implicitly when a request does not sp>
 method=GET url=/v1/acl/policy/name/nomad-server from=127.0.0.1:39550 error="Requested policy does not exist: ACL not found"
 method=GET url=/v1/acl/policy/name/nomad-client from=127.0.0.1:39558 error="Requested policy does not exist: ACL not found"
 
 ```
ubuntu@dev-node-ifca-0:~$ consul members
ubuntu@dev-node-ifca-0:~$ consul operator raft list-peers
Error getting peers: Failed to retrieve raft configuration: Unexpected response code: 403 (Permission denied: anonymous token lacks permission 'operator:read'. The anonymous token is used implicitly when a request does not specify a token.)
```
```
ubuntu@dev-node-ifca-1:~$ consul members
Node             Address             Status  Type    Build   Protocol  DC            Partition  Segment
dev-node-ifca-0  172.16.44.17:8301   alive   server  1.17.1  2         ai4os-consul  default    <all>
dev-node-ifca-1  172.16.44.163:8301  alive   client  1.17.1  2         ai4os-consul  default    <default>
dev-node-ifca-2  172.16.44.212:8301  alive   client  1.17.1  2         ai4os-consul  default    <default>
dev-node-ifca-3  172.16.44.169:8301  alive   client  1.17.1  2         ai4os-consul  default    <default>
ubuntu@dev-node-ifca-1:~$ nomad node status
ID        Node Pool  DC            Name             Class   Drain  Eligibility  Status
99de90f6  default    ifca-ai4eosc  dev-node-ifca-2  <none>  false  eligible     ready
48d5a598  default    ifca-ai4eosc  dev-node-ifca-1  <none>  false  eligible     ready
38fa7047  default    ifca-ai4eosc  dev-node-ifca-3  <none>  false  eligible     ready
ubuntu@dev-node-ifca-1:~$ 

```

**Comment by micafer** [comment]
May be we need to make a call to try to debug this

**Comment by Sftobias** [comment]
Sure

**Comment by Sftobias** [comment]
Hi Miguel,
I think ive found the error. This task is not being executed since the [hostfiles](https://github.com/ai4os/ai4-ansible/blob/master/hosts_ifca_admin_deploy_template
) of the "from scratch" deployments dosnt have the consul and nomad servers section.

```
- name: "Set agent token"
  command: consul acl set-agent-token agent "{{ agent_consul_token }}"
  changed_when: false
  when: ("consul_servers" in groups and inventory_hostname in groups["consul_servers"])
```
We should either change the condition to consul master, or **add the missing section on the hosts file**.

Im gonna reset my test machines so i can execute the roles from scratch with this change. I will post the results

File: roles/consul/tasks/set_acl_token.yml [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-    path: "{{ consul_config }}server.hcl"
-    replace: "{{ agent_consul_token }}"
-    regexp: '{{ default_token }}'
+- name: "Copy acl.hcl configuration"
+  template:
+   src: templates/consul_acl.j2
+   dest: "{{ consul_config }}acl.hcl"
+   owner: "{{ consul_username }}"
+   force: no
   notify: Restart consul
```
Comment (Sftobias) [inline roles/consul/tasks/set_acl_token.yml:8]: Why is this file needed? When the "Set agent token" task executes it creates a file in `/opt/consul` called `acl-tokens.json` wich contains the token passed to the  `consul acl set-agent-token agent` command. This file is the one used to obtain the token when needed.
